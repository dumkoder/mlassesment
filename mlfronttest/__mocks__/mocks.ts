interface itemsTypes{ 
    author: {
        name: string,
        lastName: string
    },
    categories: [string],
    items: [{
      id: string,
      picture: string,
      title: string,
      free_shipping: boolean,
      condition: string,
      price:{
        amount: number,
        currency: string, 
        decimals: number
      }
    }]
}

export const mockedItems: itemsTypes = {
    author:{
            name:"MockAuthorName",
            lastName:"mockAuthorLastName"
        },
        categories:[
            "categoryOne"
        ],
        items:[{
          id:"MLA884674542",
          title:"Mock Title",
          price:{
              currency:"ARS",
              amount:6839,
              decimals:0
          },
          picture:"mockImg.jpg",
          condition:"new",
          free_shipping:true
      }]
};
