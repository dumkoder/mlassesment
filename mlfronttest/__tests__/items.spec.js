import Items, {getServerSideProps} from '../pages/items'
import { render, screen, waitFor } from '@testing-library/react'
import { mockedItems } from '../__mocks__/mocks';

describe('Render Page Items',  () => {
  beforeEach( async ()=>{
    global.fetch = jest.fn(() => Promise.resolve({ json: () => Promise.resolve({mockedItems})}))
    await getServerSideProps({query: { search: "mockText" }});
  })
  
  test('should get Items form props', async () => {
    const comp = render(<Items results={mockedItems}/>)

    expect(comp).toMatchSnapshot()
      
  });
});