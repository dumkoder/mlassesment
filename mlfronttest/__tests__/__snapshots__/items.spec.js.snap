// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Page Items should get Items form props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="container"
        >
          <a
            class="logo"
            href="/"
          >
            <img
              alt="Logo Mercadolibre"
              src="/Logo_ML.png"
              srcset="/Logo_ML@2x.png 2x"
            />
          </a>
          <form
            method="GET"
            role="searchForm"
          >
            <input
              id="search"
              name="search"
              placeholder="Nunca dejes de buscar"
              required=""
              type="text"
              value=""
            />
            <button
              type="submit"
            >
              <img
                alt="Buscar"
                src="/ic_Search.png"
                srcset="/ic_Search@2x.png 2x"
              />
            </button>
          </form>
        </div>
      </header>
      <div
        class="container"
      >
        <ul>
          <li>
            <p>
              categoryOne
            </p>
          </li>
        </ul>
      </div>
      <main
        class="container"
      >
        <ul>
          <li>
            <article>
              <div
                class="thumbImage"
              >
                <img
                  alt="Mock Title"
                  height="180"
                  src="mockImg.jpg"
                  width="180"
                />
              </div>
              <div
                class="description"
              >
                <div
                  class="price"
                >
                  <p>
                    $ 6.839,00
                  </p>
                  <img
                    alt="envio gratis"
                    src="ic_shipping.png"
                    srcset="ic_shipping@2x.png 2x"
                  />
                </div>
                <h2>
                  <a
                    href="/items/MLA884674542"
                  >
                    Mock Title
                  </a>
                </h2>
              </div>
              <div
                class="condition"
              >
                new
              </div>
            </article>
          </li>
        </ul>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="container"
      >
        <a
          class="logo"
          href="/"
        >
          <img
            alt="Logo Mercadolibre"
            src="/Logo_ML.png"
            srcset="/Logo_ML@2x.png 2x"
          />
        </a>
        <form
          method="GET"
          role="searchForm"
        >
          <input
            id="search"
            name="search"
            placeholder="Nunca dejes de buscar"
            required=""
            type="text"
            value=""
          />
          <button
            type="submit"
          >
            <img
              alt="Buscar"
              src="/ic_Search.png"
              srcset="/ic_Search@2x.png 2x"
            />
          </button>
        </form>
      </div>
    </header>
    <div
      class="container"
    >
      <ul>
        <li>
          <p>
            categoryOne
          </p>
        </li>
      </ul>
    </div>
    <main
      class="container"
    >
      <ul>
        <li>
          <article>
            <div
              class="thumbImage"
            >
              <img
                alt="Mock Title"
                height="180"
                src="mockImg.jpg"
                width="180"
              />
            </div>
            <div
              class="description"
            >
              <div
                class="price"
              >
                <p>
                  $ 6.839,00
                </p>
                <img
                  alt="envio gratis"
                  src="ic_shipping.png"
                  srcset="ic_shipping@2x.png 2x"
                />
              </div>
              <h2>
                <a
                  href="/items/MLA884674542"
                >
                  Mock Title
                </a>
              </h2>
            </div>
            <div
              class="condition"
            >
              new
            </div>
          </article>
        </li>
      </ul>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
